### PR TITLE
refactor(query): add more context to "Missing queryFn" error

### PR DESCRIPTION
**Pull request by GLabat**
For now, we display the queryKey for which a queryFn is missing.

### CODE REVIEW

File: packages/query-core/src/infiniteQueryBehavior.ts [hunk 1]
```diff
@@ -44,7 +44,11 @@ export function infiniteQueryBehavior<
 
         // Get query function
         const queryFn =
-          context.options.queryFn || (() => Promise.reject('Missing queryFn'))
+          context.options.queryFn ||
+          (() =>
+            Promise.reject(
+              `Missing queryFn for queryKey '${context.queryKey}'`,
```
[the remark this answers is not in the document]
Comment (GLabat) [reply packages/query-core/src/infiniteQueryBehavior.ts:50]: Yep sure. Should we take care of custom hash function?

[the remark this answers is not in the document]
Comment (TkDodo) [reply packages/query-core/src/infiniteQueryBehavior.ts:50]: no, the `queryHash` is already produced from the custom hashFn or the default one

File: packages/query-core/src/tests/query.test.tsx [hunk 2]
```diff
@@ -794,8 +794,12 @@ describe('query', () => {
     })
 
     const unsubscribe = observer.subscribe(() => undefined)
+
     await sleep(10)
-    expect(mockLogger.error).toHaveBeenCalledWith('Missing queryFn')
+    const query = queryCache.find(key)!
+    expect(mockLogger.error).toHaveBeenCalledWith(
+      `Missing queryFn for queryKey '${query.queryHash}'`,
```
Comment (GLabat) [inline packages/query-core/src/tests/query.test.tsx:801]: @TkDodo question here: wondering if testing the hash is the right thing to do here. It is an implementation detail. What about testing partially on `Missing queryFn for queryKey`?

WDYT?
Thanks.

Comment (TkDodo) [reply packages/query-core/src/tests/query.test.tsx:801]: just theoretically, the test would then pass even if we don't put the hash in there. I think testing for it is fine, even with a different hashFn, you would see the desired result

File: packages/query-core/src/infiniteQueryBehavior.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (GLabat) [reply packages/query-core/src/infiniteQueryBehavior.ts]: Should be ok now. I don't know what to do with the failing test. All tests pass locally, including the react17 one.

**Review by TkDodo — APPROVED** [review]

**Pull request merged**